incubator/hnc/internal/reconcilers/hierarchy_config_test.go

```diff
@@ -135,9 +131,105 @@ var _ = Describe("Hierarchy", func() {
 		setParent(ctx, barName, fooName)
 
 		// Verify all the labels
-		Eventually(getLabel(ctx, barName, fooName+depthSuffix)).Should(Equal("1"))
-		Eventually(getLabel(ctx, barName, barName+depthSuffix)).Should(Equal("0"))
-		Eventually(getLabel(ctx, fooName, fooName+depthSuffix)).Should(Equal("0"))
+		Eventually(getLabel(ctx, barName, fooName+api.LabelTreeDepthSuffix)).Should(Equal("1"))
+		Eventually(getLabel(ctx, barName, barName+api.LabelTreeDepthSuffix)).Should(Equal("0"))
+		Eventually(getLabel(ctx, fooName, fooName+api.LabelTreeDepthSuffix)).Should(Equal("0"))
+	})
+
+	It("should propagate external tree labels", func() {
+		// Create an external namespace baz
+		l := map[string]string{
+			"ext2" + api.LabelTreeDepthSuffix: "2",
+			"ext1" + api.LabelTreeDepthSuffix: "1",
+		}
+		a := map[string]string{api.AnnotationManagedBy: "others"}
+		bazName := createNSWithLabelAnnotation(ctx, "baz", l, a)
+
+		// Make bar a child of baz
+		setParent(ctx, barName, bazName)
+
+		// Verify all the labels
+		Eventually(getLabel(ctx, barName, "ext2"+api.LabelTreeDepthSuffix)).Should(Equal("3"))
+		Eventually(getLabel(ctx, barName, "ext1"+api.LabelTreeDepthSuffix)).Should(Equal("2"))
+		// Even if baz doesn't have the tree label of itself with depth 0 when created,
+		// the tree label of itself is added to the forest automatically.
+		Eventually(getLabel(ctx, barName, bazName+api.LabelTreeDepthSuffix)).Should(Equal("1"))
+		Eventually(getLabel(ctx, barName, barName+api.LabelTreeDepthSuffix)).Should(Equal("0"))
+
+		Eventually(getLabel(ctx, bazName, "ext2"+api.LabelTreeDepthSuffix)).Should(Equal("2"))
+		Eventually(getLabel(ctx, bazName, "ext1"+api.LabelTreeDepthSuffix)).Should(Equal("1"))
+		Eventually(getLabel(ctx, bazName, bazName+api.LabelTreeDepthSuffix)).Should(Equal(""))
```

Comment:
Why is this the empty string and not "0"?

Reply:
Updated the logic and thus having "0" here now.